bugsnag/src/main/java/com/bugsnag/Configuration.java

```diff
@@ -46,12 +47,17 @@ public class Configuration {
     public boolean sendThreads = false;
 
     Collection<Callback> callbacks = new ConcurrentLinkedQueue<Callback>();
-    Serializer serializer = new Serializer();
+    private ISerializer serializer;
     private final AtomicBoolean autoCaptureSessions = new AtomicBoolean(true);
     private final AtomicBoolean sendUncaughtExceptions = new AtomicBoolean(true);
 
     Configuration(String apiKey) {
+        this(apiKey, new Serializer());
+    }
+
+    Configuration(String apiKey, ISerializer serializer) {
```

Comment:
We only have API key as a parameter for configuration - all the others should be initialised with a default (i.e. the default Serializer implementation) and be set via properties.